### PR TITLE
feat: add kube-monkey, metrics-server, loki, prometheus, cilium, aws-node-termination-handler

### PR DESCRIPTION
**Pull request by qvistgaard**

### CODE REVIEW

**Comment by anarsen** [comment]
Please reword commit 4cd60af to be more descriptive. These will be in the changelog, so it will be useful to have it state what exactly is added.

Perhaps the modules should be in individual commits to really make it clear. That's not too important right now tbh.

And the linting is blocking the merge.

File: CODEOWNERS
Comment (anarsen) [inline CODEOWNERS]: These are not relevant for this repository. I can remove them if you want?

**Comment by qvistgaard** [comment]
I agree that each module probable should have been multiple commits.

File: CODEOWNERS
Comment (qvistgaard) [reply CODEOWNERS]: sure go ahead

File: modules/cilium/main.tf
Comment (anarsen) [inline modules/cilium/main.tf]: Cluster name must be provided through module variables.

File: modules/cilium/main.tf
Comment (anarsen) [inline modules/cilium/main.tf]: We've set `wait = true` in the other add-ons. I like the thought of everything being in a good state when Terraform completes, but I'd like to know what you think since you set it `false`.

File: modules/cilium/variables.tf
Comment (anarsen) [inline modules/cilium/variables.tf]: Variable should be more descriptive imo.

File: modules/cilium/variables.tf
Comment (anarsen) [inline modules/cilium/variables.tf]: Variable should be more descriptive imo.

File: modules/kube-monkey/scripts/download-helm-chart.sh
Comment (anarsen) [inline modules/kube-monkey/scripts/download-helm-chart.sh:1]: Is this script actually being referenced?

**Review by anarsen — CHANGES_REQUESTED** [review]
I don't like it that we're littering `kube-system` these components tbh, but you know that :stuck_out_tongue: 

I'd prefer if the add-on modules under `modules/` were more alike. E.g. the use of `locals` for storing values for `helm_release` to use - you know my stance on that as well.

In spite of that, I think we should just get this through and iterate over all of that later on. My requests for changes in this review still stand though :slightly_smiling_face:

File: modules/cilium/main.tf
Comment (qvistgaard) [reply modules/cilium/main.tf]: why?

Comment (wolffberg) [reply modules/cilium/main.tf]: We should only deploy the charts into existing clusters, so the name should never be random.

File: modules/kube-monkey/scripts/download-helm-chart.sh
Comment (qvistgaard) [reply modules/kube-monkey/scripts/download-helm-chart.sh:1]: no, but you use it if you want to update the helm chart. I don't want it updated each time terraform is run

File: CODEOWNERS
Comment (anarsen) [inline CODEOWNERS]: @qvistgaard I think you overwrite my commit with the changes I made to this file.

Anyway, could you move lines 8-10 to `.github/CODEOWNERS` and delete this file altogether?

File: modules/aws-node-termination-handler/main.tf
Comment (anarsen) [inline modules/aws-node-termination-handler/main.tf:16]: nitpick: why keep _some_ of the values as `locals` while `create_namespace` and `wait` are set directly on the `helm_release`?

**Comment by anarsen** [comment]
I'm taking the liberty of squashing a few of the commits. E.g. all the formatting and alterations made to other cilium and kube-monkey in this PR are squashed into the commit actually _adding_ the modules.

**Comment by anarsen** [comment]
This is fine for me to accept now. What about you @wolffberg?

Just a reminder for us to discuss the whole `kube-system` once and for all at another time: #12

**Comment by anarsen** [comment]
:information_source: If any of you decide to merge, be sure to select _Rebase and merge_ to ensure a linear history on `master`. The commit messages and scope are in a good shape now.

**Comment by anarsen** [comment]
@qvistgaard I'm taking the liberty of squashing a few of the commits. Commits such as ` feat: Added namespace variable to loki` will appear in the changelog as feature additions, which this isn't really. It's just a part of the `feat: add loki` commit.

**Review by anarsen — APPROVED** [review]

**Pull request merged**
